app/react/Layout/TableViewer.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { Component } from 'react';
+
+import { connect } from 'react-redux';
+import { TableRow } from 'app/Library/components/TableRow';
+
+export interface DocumentViewerProps {
+  rowListZoomLevel: number;
+  documents: any;
+  storeKey: 'library' | 'uploads';
+  clickOnDocument: (...args: any[]) => any;
+  onSnippetClick: (...args: any[]) => any;
+  deleteConnection: (...args: any[]) => any;
+  search: any;
+  templates: any;
+  thesauris: any;
+  columns: any[];
+}
+
+class TableViewerComponent extends Component<DocumentViewerProps> {
+  render() {
+    const columns = this.props.columns.filter(c => !c.get('hidden'));
+    return (
+      <div className="tableview-wrapper">
+        <table>
+          <thead>
+            <tr>
+              {columns.map((column: any, index: number) => (
+                <th className={!index ? 'sticky-col' : ''} key={index}>
```

Comment:
This is the precise reason why our rules prevent using 'index' as key.  Here, columns can change, but the index will remain the same.  Please, find a way to use a different key.  Perhaps column name is disambiguation enough?